hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -814,18 +820,22 @@ private void validateSchema() throws HoodieUpsertException, HoodieInsertExceptio
   }
 
   public void validateUpsertSchema() throws HoodieUpsertException {
-    try {
-      validateSchema();
-    } catch (HoodieException e) {
-      throw new HoodieUpsertException("Failed upsert schema compatibility check", e);
+    if (!isMetadataTable) {
+      try {
+        validateSchema();
+      } catch (HoodieException e) {
+        throw new HoodieUpsertException("Failed upsert schema compatibility check", e);
+      }
     }
   }
 
   public void validateInsertSchema() throws HoodieInsertException {
-    try {
-      validateSchema();
-    } catch (HoodieException e) {
-      throw new HoodieInsertException("Failed insert schema compatibility check", e);
+    if (!isMetadataTable) {
+      try {
+        validateSchema();
+      } catch (HoodieException e) {
+        throw new HoodieInsertException("Failed insert schema compatibility check", e);
+      }
     }
   }
```

Comment:
We have evolved the metadata payload schema before and there were cases where the schema change is not backward compatible.  Have you tested if disabling schema validation is going to swallow the issue and cause corruption of the metadata table?

Reply:
my take here is. when we evolve the schema(anyways we will put up a new patch for it and is controlled), we can enable schema validation once and validate its compatible.

Reply:
have added a jira to track adding a test for this https://issues.apache.org/jira/browse/HUDI-5566